spec/classes/puppet_agent_spec.rb

```diff
@@ -203,7 +203,9 @@ def global_facts(facts, os)
 
         context 'package_version is same as master when set to auto' do
           let(:params) { { package_version: 'auto' } }
-          let(:node_params) { { serverversion: '7.6.5' } }
+          let(:facts) do
+            global_facts(facts, os).merge(serverversion: '7.6.5')
+          end
```

Comment:
I was getting test failures after the PDK update relating to this line and variable reassignment (see failed tests here: https://github.com/puppetlabs/puppetlabs-puppet_agent/actions/runs/7481505671/job/20363217135).

I assumed it was because we were treating this as a parameter instead of a fact. I changed it to what you see here and tests are passing again, but I'm not 100% certain that this is the right approach.

Reply:
Looks to match the behavior below so :+1: to me: 

```ruby
        let(:facts) do
          global_facts(facts, os).merge(is_pe: true)
        end
```